lib/reversetunnel/srv.go

```diff
@@ -1163,18 +1174,26 @@ func sendVersionRequest(ctx context.Context, sconn ssh.Conn) (string, error) {
 			errorCh <- trace.BadParameter("no response to %v request", versionRequest)
 			return
 		}
-		versionCh <- string(payload)
+		var response versionResponse
```

Comment:
What if a remote tunnel agent uses a new json format for version reporting and  client still uses old `sendVersionRequest` logic ?.  In that case the whole json formatted version payload will be interpreted as raw string teleport version.

Reply:
Good point. Since it looks like this is only being used to determine the remote site cache policy I think removing the new json message and returning the auth version instead of the agent version makes more sense. What do you think?

Reply:
Implemented this in https://github.com/gravitational/teleport/pull/12130/commits/134451a7cb31604a35040d8f702841df910166d3